Zend/tests/float_to_int/explicit_casts_should_not_warn.phpt

```diff
@@ -0,0 +1,34 @@
+--TEST--
+Explicit (int) cast must not warn
+--FILE--
+<?php
+
+$values =[
+    3.0,
+    3.5,
+    10e120,
+    10e300,
+    fdiv(0, 0),
+    (string) 3.0,
+    (string) 3.5,
+    (string) 10e120,
+    (string) 10e300,
+    (string) fdiv(0, 0),
+];
+
+foreach($values as $value) {
+    var_dump((int) $value);
+}
+
+?>
+--EXPECT--
+int(3)
+int(3)
+int(0)
+int(0)
+int(0)
+int(3)
+int(3)
+int(9223372036854775807)
```

Comment:
Should this test be skipped on 32-bit builds (`PHP_INT_SIZE === 4`)? Or maybe print PHP_INT_MAX as a string instead of calling var_dump if it would result in PHP_INT_MAX

See tests/lang/operators/bitwiseNot_basiclong_64bit.phpt

```
--SKIPIF--
<?php
if (PHP_INT_SIZE != 8) die("skip this test is for 64bit platform only");
?>
```